### PR TITLE
Us133252/csv export

### PR DESCRIPTION
**Pull request by johngwilkinson**
This adds csv export which creates a .zip from all dataSets which are marked for export. So far, only System Access is actually exported.

Note: switched to csv-stringify from export-from-json because it saves having to construct full json objects with header names that just get converted back to plain csv rows again. We also no longer can use the save-file functionality of export-from-json because we need to build a .zip from all the csvs.

Quality Notes
- functional
  - demo page: export and check csv; also check with multiple files in csv
  - [x] e2e
- performance: n/a
- security: n/a
- devops: n/a
- ux/docs: same button as other dashboards

### CODE REVIEW

File: components/cards/system-access.js [hunk 1]
```diff
@@ -211,13 +211,7 @@ class SystemAccessCard extends SkeletonMixin(Localizer(MobxLitElement)) {
...
 
 	get data() {
-		// possibly the server will return timestamps and not be nested, in which case this may need to adjust
-		// for the server timezone
-		return this.resultSet.data.systemAccessData?.map(([d, v]) => {
-			const [year, month, day] = d.split('-');
-			// midnight, browser time
-			return [new Date(year, month - 1, day, 0, 0, 0).getTime(), v];
-		}) || [];
+		return this.dataSet.data;
```
Comment (Vitalii-Misechko) [inline components/cards/system-access.js:214]: nice!

File: model/dataSet.js [hunk 2]
```diff
@@ -18,6 +18,29 @@ export class DataSet {
 		return data;
 	}
 
+	get isExport() {
+		return false;
+	}
```
Comment (Vitalii-Misechko) [inline model/dataSet.js:23]: Should we move all these new members to a mixin?

File: model/dataSets/systemAccessDataSet.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+			// midnight, browser time
+			return [new Date(year, month - 1, day, 0, 0, 0).getTime(), v];
+		}) || [];
+	}
+
+	get isExport() { return true; }
+	processForExport(data) { return data.systemAccessData || []; }
+	get headersForExport() { return ['export:date', 'export:systemAccessCount']; }
+	get exportFilenameTerm() { return 'export:systemAccessFilename'; }
+}
```
Comment (Vitalii-Misechko) [inline model/dataSets/systemAccessDataSet.js:16]: neat!

File: model/data.js [hunk 4]
```diff
@@ -12,11 +13,15 @@ export class Data {
 		this._resultSets.forEach(resultSet => resultSet.load());
 	}
 
-	getResultSet(name) {
-		return this._resultSets.find(r => r.name === name);
+	getResultSet(resultSet) {
```
Comment (Vitalii-Misechko) [inline model/data.js:16]: `resultSet` => `resultSetInfo`?

File: model/resultSets.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
 export const resultSets = {
```
Comment (Vitalii-Misechko) [inline model/resultSets.js:1]: `resultSets` => `resultSetInfos`?

File: model/data.js [hunk 6]
```diff
@@ -12,11 +13,15 @@ export class Data {
...
+		return this._resultSets.find(r => r.name === resultSet.name);
 	}
 
 	getDataSet(name) {
 		return this._dataSets.get(name);
 	}
+
+	async export(localize) {
+		await exportToZip(localize('export:filename'), [...this._dataSets.values()].filter(d => d.isExport), localize);
+	}
```
Comment (Vitalii-Misechko) [inline model/data.js:26]: `thought` It feels like this implementation can be a part of `adoption-dashboard-view.js` or a separate class that accesses `_dataSets` via getter. But it's fine here I guess

**Review by Vitalii-Misechko — APPROVED** [review]
LGTM!

File: locales/en.js [hunk 7]
```diff
@@ -22,5 +23,11 @@ export default {
...
 
 	"toolUsageTrendCard:title": "Tool Usage Trend by Tool",
-	"toolUsageTrendCard:yAxisTitle": "Usage Count"
+	"toolUsageTrendCard:yAxisTitle": "Usage Count",
+
+	"export:filename": "Adoption Dashboard",
+	"export:date": "Date",
+	"export:systemAccessCount": "System Access Count",
+	"export:systemAccessFilename": "Adoption Dashboard - 'System Access'",
+	"export:toolUsageTrendFilename": "Adoption Dashboard - 'Tool Usage Trend by Tool'"
```
Comment (bemailloux) [inline locales/en.js:32]: Should the file names be more "computer friendly" (like camel or snake case)?

**Review by bemailloux — APPROVED** [review]

File: locales/en.js (hunk 7 above)
Comment (johngwilkinson) [reply locales/en.js:32]: These names were specified by PM. Do you have specific concerns about them that I should take to her?

Comment (bemailloux) [reply locales/en.js:32]: Just thinking about that one lab computer every institution has that's still running XP or older. But usually those are "off-the-grid" these days :)

**Pull request merged**